crates/cxx-qt-lib/src/core/qanystringview.cpp

```diff
@@ -0,0 +1,47 @@
+// clang-format off
+// SPDX-FileCopyrightText: 2024 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// clang-format on
+// SPDX-FileContributor: Joshua Goins <joshua.goins@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+#include "cxx-qt-lib/qanystringview.h"
+
+#include "../assertion_utils.h"
+
+// QAnyStringView has two members.
+// A union of (void*, char*, char_16*) and a size_t.
+// https://code.qt.io/cgit/qt/qtbase.git/tree/src/corelib/text/qanystringview.h
+assert_alignment_and_size(QAnyStringView,
+                          alignof(::std::size_t),
+                          sizeof(::std::size_t) + sizeof(void*));
+
+static_assert(::std::is_trivially_copy_assignable<QAnyStringView>::value);
+static_assert(::std::is_trivially_copy_constructible<QAnyStringView>::value);
+
+static_assert(::std::is_trivially_destructible<QAnyStringView>::value);
```

Comment:
i assume it is not `is_trivially_copyable` ? As if it was then you can derive Clone etc. But i assume not as there are pointers here internally.

Then it is trivially copy constructable/assignable and destructible so it needs a Clone implementation but does it also need a Drop implementation ? We don't seem to have an existing type like this :-) cc @LeonMatthesKDAB

Reply:
Aha i see there is no deconstructor :-D